### ISSUE TITLE
Decompiler produces broken if statements

### ISSUE THREAD

**Issue by sussyGaymer** [j4k0xb/View8#5]
The decompiler produced the following:
```
		if (!r13(r15))
			|| (!r6[Scope[3][2](610)])
		r13 = r6[0]
			|| (r13 == "")
```
It's pretty self-explanatory what's wrong. I'll try to find and provide the underlying bytecode.

**Comment by sussyGaymer** [comment]
I also noticed things like:
```
					if (!r15 === 0)
					{
						throw r14
						ACCU = process["exit"](1)
					}
```
(code after throw/return statements)

**Comment by sussyGaymer** [comment]
Update: After a second thought, the initial code makes *some* sense, but if statement conditions are still being closed after the 1st condition in the boolean chain.
```
		if (!user)

		r5 = typeof(user)

			|| (!r5 === Scope[8][2](286))

		{

			r0 = !([])

		}
```
This means something like:
```
if (!user || (typeof(user) !== Scope[8][2](286)) {
    r0 = false
}
```

**Comment by j4k0xb** [comment]
all upstream issues
the decompiler output is based on string concatenation/replacements, which inevitably results in broken code in some cases
I can only attempt fixing some of it (wrap `if (!r15 === 0)` condition in `()`), not solve the root issue without rewriting everything from scratch to use an AST

**Comment by sussyGaymer** [comment]
Moving to AST sounds like a lot of work... Do as you please.

**Comment by j4k0xb** [comment]
yeah it will be a lot of work... pretty much all decompilers use that + control/data flow analysis
So I'd rather not spend too much time maintaining this fork or trying to make it do things it wasn't designed for

However if you want to help fix the issues you mentioned: PR welcome 😅
